### PR TITLE
Add Great Beaked Toad

### PR DESCRIPTION
**Pull request by TheSaddestGoomba**
#### Summary
Content "Adds an evolution of giant toads with an armor-piercing and venomous bite."

#### Purpose of change
To add more distinct monsters to the amphibian grouping. The great beaked toads are a tougher, slower giant toad with an armor-piercing and venomous bite.
Currently they appear by growing from mutant toads and zombify into gastro bufo, though I intend to add a unique zombie version in a future PR.

#### Testing
I have spawned the creature both through debug spawning and advancing time on naturally spawned toads.
Instigated several test fights between it and other creatures, NPCs, and PCs.

### CODE REVIEW

File: data/json/monsters/reptile_amphibian.json [hunk 1]
```diff
@@ -391,6 +392,47 @@
...
+        "accuracy": 6,
+        "damage_max_instance": [
+          { "damage_type": "cut", "amount": 16, "armor_multiplier": 0.3 },
+          { "damage_type": "bash", "amount": 8, "armor_multiplier": 0.3 }
+        ]
+      },
+      { "type": "leap", "cooldown": 10, "move_cost": 0, "max_range": 10, "min_consider_range": 2 }
+    ],
+    "anger_triggers": [ "STALK", "PLAYER_WEAK", "PLAYER_CLOSE" ],
+    "flags": [ "SEES", "SMELLS", "HEARS", "SWIMS", "GRABS", "VENOM" ],
```
Comment (Venera3) [inline data/json/monsters/reptile_amphibian.json:435]: `VENOM` is very underpowered, and is applied to every attack by the critter. Use one of the JSON venom effects on the bite itself.

File: data/json/monsters/reptile_amphibian.json [hunk 2]
```diff
@@ -391,6 +392,47 @@
...
+    "volume": "150 L",
+    "weight": "200 kg",
+    "hp": 200,
+    "speed": 80,
+    "symbol": "B",
+    "color": "brown",
+    "aggression": 10,
+    "morale": 100,
+    "melee_skill": 5,
+    "melee_dice": 2,
```
Comment (Venera3) [inline data/json/monsters/reptile_amphibian.json:416]: Don't lean into bump damage this much. If they scratch add a scratch to them, if they bite they should bite often, the base damage is only for undirected bashing and should be lower than their actual attacks.

File: data/json/monsters/reptile_amphibian.json [hunk 3]
```diff
@@ -341,8 +341,9 @@
     "id": "mon_toad",
     "type": "MONSTER",
     "name": { "str": "toad" },
-    "description": "Toads are a diverse and varied family of amphibians sister to frogs.  Very distinguishable for their warty and dry skin.",
+    "description": "Toads are a largely colloquial grouping of frogs.  Distinguished by their warty and dry skin.",
     "copy-from": "mon_frog",
+    "upgrades": { "half_life": 34, "into": "mon_toad_small" },
```
Comment (Venera3) [inline data/json/monsters/reptile_amphibian.json:346]: Don't use straight evolution for mutants - that's a zombie thing. Either set up the groups to spawn them occasionally later and/or do a `base->egg->base or weird` progression in their reproduction.

**Review by Venera3 — CHANGES_REQUESTED** [review]
More critters are always nice, but they need some polish.

File: data/json/monsters/reptile_amphibian.json (hunk 3 above)
Comment (TheSaddestGoomba) [reply data/json/monsters/reptile_amphibian.json:346]: I'm uncertain how to implement this. The "upgrades" line was added to mimic the same in the existing "mon_frog" entry, and also overwrite its "into" value from the "copy-from."
It looks as though the existing .json had (normal) toads and frogs both upgrading to mutant bullfrogs. My intent was to redirect the toads to their own existing mutant versions and add the beaked toad to the end of that line, akin to the frog mutants:
tadpole -> frog -> mutant bullfrog -> giant bullfrog -> colossal croaker
brown tadpole -> toad -> huge toad -> giant toad -> great beaked toad

I am looking into adding the "mon_toad_beak" to group spawns as well

Comment (Venera3) [reply data/json/monsters/reptile_amphibian.json:346]: If you use evolution they won't exist as tadpoles ever - evo clock starts at game start, so after X time fully-formed mutanta would pop out of eggs. If you want them to *grow* you want to use `age_grow`, and handle world progression by their relevant spawn monster groups.

For a reference of what I mean check out the cellar spiders as an example.

**Comment by TheSaddestGoomba** [comment]
I performed a simple test, just copying over the JSON file and loading up a fresh world. I used debug to teleport around and progress the game clock. All variants appear to spawn in the correct locations and 'younger' variants still spawn from tadpoles before aging into the larger forms.

**Comment by TheSaddestGoomba** [comment]
Closing this to reimplement in a future PR after #67259 is merged.

**Pull request closed without merging**